tests/integration/__init__.py

```diff
@@ -108,6 +155,38 @@ def cleanup_runtime_config_instance(to_cleanup):
 
 atexit.register(cleanup_runtime_config_instance, RUNTIME_CONFIGS)
 
+_RUNTESTS_PORTS = {}
+
+
+def get_unused_localhost_port():
+    '''
+    Return a random unused port on localhost
+    '''
+    usock = socket.socket(family=socket.AF_INET, type=socket.SOCK_STREAM)
+    usock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    usock.bind(('127.0.0.1', 0))
+    port = usock.getsockname()[1]
+    if port in (54505, 54506, 64505, 64506, 64510, 64511):
+        # These ports are hardcoded in the test configuration
+        usock.close()
+        return get_unused_localhost_port()
```

Comment:
@s0undt3ch : Even if you close the port after calling `get_unused_localhost_port()`, you can still run into some race conditions, such as if you first pick port 54505 and then `get_unused_localhost_port()` could pick the next port, 54506 (which is also reserved).

The more robust solution is to use reserved ports outside the ephemeral range to avoid collisions with other processes. A slightly worse solution is to open up all the reserved ports at startup (and abort your tests if another process already grabbed one of them).